File: test/ayush.dev/scala/pipelines/ScalaInsideProjectDatasetsOnly/code/src/main/scala/io/prophecy/pipelines/scalainsideprojectdatasetsonly/udfs/UDFs.scala
package io.prophecy.pipelines.scalainsideprojectdatasetsonly.udfs

import _root_.io.prophecy.abinitio.ScalaFunctions._
import _root_.io.prophecy.libs._
import org.apache.spark.sql.types._
import org.apache.spark.sql.functions._
import org.apache.spark.sql._

object UDFs extends Serializable {
  var int_val = 2

  def registerUDFs(spark: SparkSession) =
    spark.udf.register("intudf", intudf)

  def intudf = udf((value: Int) => value * value / int_val)
}

File: test/ayush.dev/python/pipelines/python_external_git_pipeline/code/python_external_git_pipeline/udfs/UDFs.py
from pyspark.sql import *
from pyspark.sql.functions import *
from pyspark.sql.types import *
from prophecy.lookups import (
    createLookup,
    createRangeLookup,
    lookup,
    lookup_last,
    lookup_match,
    lookup_count,
    lookup_row,
    lookup_row_reverse,
    lookup_nth
)
int_value = 10
float_value = 20

def registerUDFs(spark: SparkSession):
    spark.udf.register("squared", squared)

@udf(returnType = IntegerType())
def squared(value):
    return ((value * value) + int_value - float_value) if value else int_value
